mm2src/coins/z_coin.rs

```diff
@@ -620,18 +625,22 @@ impl<'a> UtxoCoinWithIguanaPrivKeyBuilder for ZCoinBuilder<'a> {
         let mut db_dir_path = self.db_dir_path;
 
         db_dir_path.push(&db_name);
-        if !db_dir_path.exists() {
-            let default_cache_path = PathBuf::new().join("./").join(db_name);
-            if !default_cache_path.exists() {
-                return MmError::err(ZCoinBuildError::SaplingCacheDbDoesNotExist {
-                    path: std::env::current_dir()?.join(&default_cache_path).display().to_string(),
-                });
+        let sqlite = tokio::task::block_in_place(move || {
+            if !db_dir_path.exists() {
+                let default_cache_path = PathBuf::new().join("./").join(db_name);
```

Comment:
It's shorter, but involves nested macro. Will it also be more efficient in terms of performance? 🙂

Reply:
The performance difference would be so small, but we can consider that. If you check the execution flow for both of them, even single `join` requires more `space complexity`. And I believe it must require more `time complexity` as well.

I just suggested shorter version because it looked better to me, maybe it's just me :) Not really required change.

Reply:
I was more curious regarding performance rather than considered it's a real argument against your proposal 🙂 But since it's shorter and faster, I will make the change, thanks!